### ISSUE TITLE
NumberTokenLoss returns NaN when no number tokens are present in labels

### ISSUE THREAD

**Issue by Zuozhuo** [tum-ai/number-token-loss#73]
<img width="1242" height="510" alt="Image" src="https://github.com/user-attachments/assets/e2af939a-85a1-4a3e-a87f-b706821ef521" />

**Description**
When using [MWE of NTL](https://github.com/tum-ai/number-token-loss/blob/main/src/ntl/loss_functions/base_number_token_loss.py), if a batch of labels does not contain any number tokens, the loss returned is `NaN`.
This happens because the mask `valid_positions` is completely `False`, leading to an empty loss tensor, and calling `torch.mean` on an empty tensor results in `NaN`.

---

**Expected Behavior**

* If no numeric tokens are present in `labels`, the loss currently returns `NaN`.
* It is unclear what the correct behavior should be in this situation. Possible options include:

  * Returning `0.0` (i.e., treating it as “no contribution from NTL”),
  * Skipping NTL computation for this batch,
  * Or raising a more explicit error.
* Clarification from the authors/maintainers would be helpful.

---

**Open Question / Proposed Discussion**
Should `NumberTokenLoss` return `0.0`, raise an error, or handle this case differently when no numeric tokens are present in the labels?

**Comment by Zuozhuo** [comment]
Maybe can be this:

```
        # 🚩 robustness: if there are no number tokens → return 0
        if not valid_positions.any():
            if reduction == "none":
                return torch.zeros_like(labels, dtype=logits.dtype)
            else:
                return torch.tensor(0.0, device=logits.device, dtype=logits.dtype)
```

**Comment by jannisborn** [comment]
Hi @Zuozhuo,
Thanks for the interest! In general, NTL is always 0 if no number tokens are present.
Otherwise, as you observe, it complicates model training. The particular MWE you refer to was made by @Larspennig, maybe he can comment on this.


In general: any reason why not using the PyPI implementation? See here: https://ibm.biz/ntl-pypi-repo.
There are also docs available: [ibm.biz/ntl-docs](https://ibm.biz/ntl-docs)
The implementation in the other repo will be maintained, instead this is the development archive. If you use the PyPI implementation it handles such cases gracefully and never returns NaNs, see here: https://ai4sd.github.io/number-token-loss/api/core/#__codelineno-0-546

I consider this closed but feel free to re-open/comment
